compiler/rustc_target/src/target_features.rs

```diff
@@ -775,20 +777,20 @@ impl Target {
                 }
             }
             "arm" => {
-                // We support 2 ABIs, hardfloat (default) and softfloat.
-                if self.has_feature("soft-float") {
-                    NOTHING
-                } else {
-                    // Hardfloat ABI. x87 must be enabled.
-                    (&["fpregs"], &[])
+                // It's unclear how the `abi` and the `soft-float` target feature interact.
+                // We use the `abi` as our primary signal, and force `soft-float` to match.
+                match &*self.abi {
+                    "eabi" => (&["soft-float"], &[]),
```

Comment:
Turns out the arm-linux-androideabi target sets the `eabi` ABI, but does *not* set `+soft-float`. What does that even mean? Is it a valid combination?

Reply:
@maurer @chriswailes @mgeisler almost all our ARM `eabi` targets set `+soft-float`; the android targets seems to be the only exception (at least they are the only exception I have found so far). Is there a reason for this or is it just an oversight?

Reply:
> Turns out the arm-linux-androideabi target sets the `eabi` ABI, but does _not_ set `+soft-float`. What does that even mean? Is it a valid combination?

Yes, it is a valid combination.

It means, in essence, "pass floats via integer registers, but internal to functions, use the hardware float instructions".

Reply:
> Yes, it is a valid combination. It means, in essence, "pass floats via integer registers, but internal to functions, use the hardware float instructions".

AFAIK, the `abi` field is not forwarded to LLVM in any form. (We have `llvm_abiname` for that.) So I can't align what you are saying with what the Rust compiler does.

Reply:
...Oh, my apologies, I was only answering the abstract question because you said ABI and not `abi`. As I mentioned, the `abi` field is just a string intended to identify what ABI we are supposed to be using. Despite it being a valid form, we may not actually be implementing anything correctly here, as I have no idea what set of mystical incantations LLVM expects here (as each target backend seems to implement the idea of "a soft-float ABI may exist" slightly differently).

Reply:
My understanding was that `+soft-float,+fpregs` is now we request a softfloat ABI but using float instructions where possible. But I am not sure whether that actually works.